blaze/compute/pandas.py

```diff
@@ -145,9 +145,15 @@ def compute_up(t, lhs, rhs, **kwargs):
     return result.reset_index()[t.fields]
 
 
-@dispatch(Symbol, (DataFrameGroupBy, SeriesGroupBy))
-def compute_up(t, gb, **kwargs):
-    return gb
+pandas_structure = DataFrame, Series, DataFrameGroupBy, SeriesGroupBy
+
+
+@dispatch(Concat, pandas_structure, pandas_structure)
+def compute_up(t, lhs, rhs, _concat=pd.concat, **kwargs):
+    if not (isinstance(lhs, type(rhs)) or isinstance(rhs, type(lhs))):
+        raise TypeError('lhs and rhs must be the same type')
+
+    return _concat((lhs, rhs), axis=t.axis)
```

Comment:
I cannot decide if I should `reset_index(drop=True)` on this so that if you are using an integer index, it is still unique after the concat.